services/navet-ms/lambdas/pollUser.js

```diff
@@ -1,78 +1,106 @@
-  return;
-};
-
-function createNavetPollEventDetail(navetData) {
-  const userObj = {
-    personalNumber: navetData.PersonId.PersonNr,
-    firstName: navetData.Namn.Fornamn,
-    lastName: navetData.Namn.Efternamn,
+
+  const [requestNavetUserError, navetUser] = await to(requestNavetUser(user.personalNumber));
+  if (requestNavetUserError) {
+    return console.error('(Navet-ms)', requestNavetUserError);
+  }
+
+  const eventDetail = getNavetPollEventDetail(navetUser);
+  await putEvent(eventDetail, 'NavetPoll', 'navet.poll');
+
+  return true;
+}
+
+function getNavetPollEventDetail(navetUser) {
+  const eventDetail = {
+    personalNumber: navetUser.PersonId.PersonNr,
+    firstName: navetUser.Namn.Fornamn,
+    lastName: navetUser.Namn.Efternamn,
     address: {
-      street: navetData.Adresser.Folkbokforingsadress.Utdelningsadress2,
-      postalCode: navetData.Adresser.Folkbokforingsadress.PostNr,
-      city: navetData.Adresser.Folkbokforingsadress.Postort,
+      street: navetUser.Adresser.Folkbokforingsadress.Utdelningsadress2,
+      postalCode: navetUser.Adresser.Folkbokforingsadress.PostNr,
+      city: navetUser.Adresser.Folkbokforingsadress.Postort,
     },
-    civilStatus: navetData.Civilstand.CivilstandKod,
+    civilStatus: navetUser.Civilstand.CivilstandKod,
   };
-  return userObj;
+
+  return eventDetail;
 }
 
-async function requestNavetUser(payload, params) {
-  let err, navetClient, navetUser, parsedData;
+async function requestNavetUser(personalNumber) {
+  const ssmParams = await NAVET_PARAMS;
+  ssmParams.personalNumber = personalNumber;
 
-  // Create HTTP client
-  [err, navetClient] = await to(client(params));
-  if (!navetClient) throwError(503);
+  const [getNavetClientError, navetRequestClient] = await to(getNavetRequestClient(ssmParams));
+  if (getNavetClientError) {
+    throw getNavetClientError;
+  }
 
-  // Get user from Navet API
-  [err, navetUser] = await to(
-    request.call(navetClient, 'post', params.personpostXmlEndpoint, payload)
+  const navetPostRequestXmlPayload = parseXml(ssmParams);
+
+  const [postNavetClientError, navetUser] = await to(
+    request.call(
+      navetRequestClient,
+      'post',
+      ssmParams.personpostXmlEndpoint,
+      navetPostRequestXmlPayload
+    )
   );
-  if (err) {
-    // Parse Error Message from XML.
-    const [, parsedErrorMessage] = await to(parseErrorMessageFromXML(err.response.data));
-    throwError(err.response.status, parsedErrorMessage || null);
+  if (postNavetClientError) {
+    if (postNavetClientError.response) {
+      const [, navetResponseXmlErrorMessage] = await to(
+        parseErrorMessageFromXML(postNavetClientError.response.data)
+      );
+
+      throw navetResponseXmlErrorMessage;
+    }
+
+    throw postNavetClientError;
   }
 
-  // Parse XML data to JSON
-  [err, parsedData] = await to(parseJSON(navetUser.data));
-  if (err) throwError(500);
+  return await getNavetUserPersonPostAsJson(navetUser);
```

Comment:
getNavetPersonPost is a bit more concise for and easier to read. Do we need to know that the return value is Json?